### PR TITLE
[CORE-7535] achievemnt with graphql - prerelease

### PR DESCRIPTION
**Pull request by trtshen**
https://intersective.atlassian.net/browse/CORE-7535

## changes
- update exist core-based API to graphql for badges list

### CODE REVIEW

File: projects/v3/src/app/services/achievement.service.ts [hunk 1]
```diff
@@ -39,64 +45,87 @@ export class AchievementService {
...
+    private apolloService: ApolloService,
     private request: RequestService,
     private utils: UtilsService,
     private demo: DemoService,
   ) { }
 
-  getAchievements(order?) {
+  graphQLGetAchievements(): Observable<Achievement[]> {
+    return this.apolloService.graphQLFetch(
+      `query badges {
```
Comment (shawnm0705) [inline projects/v3/src/app/services/achievement.service.ts]: the `query badges` is not supposed to be used here. It is used for the badge project. We should use `query achievements` instead

File: projects/v3/src/app/services/achievement.service.ts [hunk 2]
```diff
@@ -39,64 +45,126 @@ export class AchievementService {
...
+   * @return  {Observable<Achievement>[]}          achievement list in badges
+   */
+  graphQLGetAchievements(
+    filter?: string,
+    type?: string,
+    active?: boolean
+  ): Observable<Achievement[]> {
+    return this.apolloService
+      .graphQLFetch(
+        `query achievements($filter: String, $type: String, $active: Boolean) {
```
Comment (trtshen) [inline projects/v3/src/app/services/achievement.service.ts]: @shawnm0705 now using `query achievements`

Comment (shawnm0705) [reply projects/v3/src/app/services/achievement.service.ts]: When we use `query achievements` in the APP, we don't need to worry about the arguments of this endpoint. They are all built for LearnFlow. We can just query it without any arguments

File: projects/v3/src/app/services/achievement.service.ts [hunk 3]
```diff
@@ -39,64 +45,126 @@ export class AchievementService {
...
+        }
+      )
+      .pipe(
+        map(
+          (res: {
+            data: {
+              badges: Achievement[];
+            };
+          }) => {
+            return res?.data?.badges || [];
```
Comment (shawnm0705) [inline projects/v3/src/app/services/achievement.service.ts]: the response of this endpoint will be `res.data.achievements`. Please merge this code to trunk and test on stage first, before merging it to the golive branch

**Review by shawnm0705 — CHANGES_REQUESTED** [review]

**Comment by trtshen** [comment]
@shawnm0705 https://github.com/intersective/app/pull/2271#discussion_r1804287418
i did what you said, but the validation didnt get through.

Screenshot 2024-10-17 at 4 04 03 pm

and i think it's fine to just keep it as it wont hurt and can be future proof too.

**Comment by trtshen** [comment]
done, @shawnm0705 please help review again. it;s ok on staging now.
Screenshot 2024-10-17 at 4 33 03 pm

**Review by shawnm0705 — APPROVED** [review]

**Pull request merged**